library/Fission/AWS/Types.hs

```diff
@@ -0,0 +1,44 @@
+module Fission.AWS.Types
+  ( Domain (..)
+  , ZoneId (..)
+  ) where
+
+import RIO
+
+import Data.Aeson
+import Data.Swagger (ToSchema (..))
+import Servant
+
+import qualified Fission.Internal.UTF8 as UTF8
+
+newtype Domain = Domain { getDomain :: Text }
```

Comment:
At risk of being pedantic, we may want to be explicit here: `DomainName`. There are other things with "domain" in the name that are part of DNS: "network domain", "domain controller", and so on